plasmapy/diagnostics/charged_particle_radiography/synthetic_radiography.py

```diff
@@ -127,22 +160,35 @@ def __init__(
         grids: AbstractGrid | Iterable[AbstractGrid],
         source: u.Quantity[u.m],
         detector: u.Quantity[u.m],
+        dt=None,
+        dt_range=None,
+        field_weighting="volume averaged",
         detector_hdir=None,
+        output_file: Path | None = None,
         verbose: bool = True,
     ) -> None:
-        # self.grid is the grid object
-        if isinstance(grids, AbstractGrid):
-            self.grids = [
-                grids,
-            ]
-        elif isinstance(grids, collections.abc.Iterable):
-            self.grids = grids
-        else:
-            raise TypeError("Type of argument `grids` not recognized.")
+        # The synthetic radiography class handles logging, so we can disable logging for the particle tracker
+        # The particle tracker class ensures that the provided grid argument has the proper type and
+        # that the necessary grid quantities are created if they are not already specified
+        save_routine = (
+            _SyntheticRadiographySaveRoutine(output_file)
+            if output_file is not None
+            else None
+        )
 
-        # self.grid_arr is the grid positions in si units. This is created here
-        # so that it isn't continuously called later
-        self.grids_arr = [grid.grid.to(u.m).value for grid in self.grids]
+        termination_condition = AllParticlesOffGridTerminationCondition(
```

Comment:
Good to make this a default, but maybe still keep this a keyword, or make `fraction_exited_threshold` a keyword? In some cases I think I've had to fiddle with that parameter to get things to run?

Reply:
So maybe something like having to specify either `termination_condition` or `fraction_exited_threshold`?

Reply:
The tracker sort of assumes this termination condition, so I would make `fraction_exited_threshold` a keyword. The current value of 0.001 is a good default.